### PR TITLE
Update hwloc to version 2.8.0

### PR DESCRIPTION
**Pull request by fwyzard**
API changes:
  - Add HWLOC_TOPOLOGY_FLAG_NO_DISTANCES, _NO_MEMATTRS and _NO_CPUKINDS
    to reduce the overhead when unneeded.
  - Add separate Read/Write Bandwidth/Latency memory attributes and
    implement them on Linux.

Backends changes:
  - NUMA nodes may now have a subtype such as DRAM, HBM, SPM, or NVM
    on heterogeneous memory platforms on Linux.
    - Add DAXType and DAXParent attributes on Linux to tell where a
      DAX device or its corresponding NUMA node come from (SPM for
      Specific-Purpose or NVM for Non-Volatile Memory).
  - Detect heterogeneous caches in hybrid CPUs on MacOS X,
    thanks to Paul Bone for the help.
  - Max frequencies are not ignored in Linux cpukinds anymore (they were
    ignored in hwloc 2.7.0), but they may be slightly adjusted to avoid
    reporting hybrid CPUs because Intel Turbo Boost Max 3.0.
    - See the documentation of environment variable HWLOC_CPUKINDS_MAXFREQ.
  - Hardwire the PCI locality of HPE Cray EX235a nodes.

Tools:
  - lstopo and other tools may now load Linux and x86 cpuid topology files
    from a tarball.
  - lstopo may now replace the P# and L# index prefixes with custom strings
    thanks to --os-index-prefix and --logical-index-prefix options.

Misc:
  - Add --disable-readme to avoid regenerating the top-level hwloc README
    file from the documentation.

Other change:
  - Enable ROCm support
  - Rework CUDA support

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
please test for el8_amd64_gcc11

**Comment by fwyzard** [comment]
please test for slc7_amd64_gcc10

**Comment by fwyzard** [comment]
please test for el8_ppc64le_gcc10

**Comment by fwyzard** [comment]
please test for el8_aarch64_gcc10

**Comment by fwyzard** [comment]
please test for slc7_amd64_gcc10

**Comment by fwyzard** [comment]
please test for el8_ppc64le_gcc10

File: hwloc.spec [hunk 1]
```diff
@@ -1,8 +1,8 @@
-### RPM external hwloc 2.7.1
-Source: https://download.open-mpi.org/release/%{n}/v2.7/%{n}-%{realversion}.tar.bz2
+### RPM external hwloc 2.8.0
+Source: https://download.open-mpi.org/release/%{n}/v2.8/%{n}-%{realversion}.tar.bz2
 
 BuildRequires: autotools
-Requires: cuda libpciaccess libxml2 numactl
+Requires: cuda libpciaccess libxml2 numactl rocm
```
Comment (smuzaffar) [inline hwloc.spec]: @fwyzard , `rocm` is only available for `x86_64`. So I guess this should be a conditional dependency i.e.
```
%ifarch x86_64
Requires: rocm
%endif
```

File: hwloc.spec [hunk 2]
```diff
@@ -15,15 +15,16 @@ Requires: cuda libpciaccess libxml2 numactl
   --enable-cpuid \
   --enable-cuda \
   --enable-nvml \
+  --enable-rsmi \
   --enable-libxml2 \
   --disable-cairo \
   --disable-doxygen \
   --enable-plugins=cuda,nvml \
+  --with-cuda=$CUDA_ROOT \
+  --with-rocm=$ROCM_ROOT \
```
Comment (smuzaffar) [inline hwloc.spec:28]: same here  (only for x86_64)

File: hwloc.spec (hunk 1 above)
Comment (fwyzard) [reply hwloc.spec]: Yes, and also the dependency on cuda should be confitional. I'll slowly make those changes.

**Comment by fwyzard** [comment]
please test for el8_ppc64le_gcc10

**Comment by fwyzard** [comment]
please test for el8_aarch64_gcc10

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
please test for el8_amd64_gcc11

**Comment by fwyzard** [comment]
please test for slc7_amd64_gcc10

**Comment by smuzaffar** [comment]
+externals
@fwyzard , this looks good to me. Do you want to perform any tests before we merge it?

**Comment by fwyzard** [comment]
> @fwyzard , this looks good to me. Do you want to perform any tests before we merge it?

No, let's merge it.

**Pull request merged**